src/ln/channel.rs

```diff
@@ -627,6 +627,19 @@ impl Channel {
 		channel_monitor.set_their_base_keys(&msg.htlc_basepoint, &msg.delayed_payment_basepoint);
 		channel_monitor.set_their_to_self_delay(msg.to_self_delay);
 
+		let their_shutdown_scriptpubkey = if let &Some(ref features) = their_local_features {
+			if features.supports_upfront_shutdown_script() {
+				match &msg.shutdown_scriptpubkey {
+					&OptionalField::Present(ref script) => {
+						if script.is_p2pkh() || script.is_p2sh() || script.is_v0_p2wsh() || script.is_v0_p2wpkh() {
```

Comment:
Hmm, shouldn't we reject channels with a shutdown script set to something other than these (or 0-length)?

Reply:
Reworked to reject channels with a shutdown script different from accepted formats if peer is signaling option. BOLT2 : "if the scriptpubkey is not in one of the above forms:
    SHOULD fail the connection.
"